packages/core/src/index.ts

```diff
@@ -97,22 +98,35 @@ class ImageService extends Service {
     })
   }
 
-  async imgUrlToBase64(url: string): Promise<string> {
-    return this.ctx
-      .http(url, { method: 'GET', responseType: 'arraybuffer', proxyAgent: '' })
-      .then((resp) => {
-        return `data:${resp.headers['content-type']};base64,${Buffer.from(resp.data).toString('base64')}`
-      })
-      .catch((err) => {
-        if (Quester.Error.is(err)) {
-          logger.warn(
-            `Request images failed with HTTP status ${err.response?.status}: ${JSON.stringify(err.response?.data)}.`,
-          )
-        } else {
-          logger.error(`Request images failed with unknown error: ${err.message}.`)
+  async imgUrlToBase64(url: string, resize: boolean = false, size: number = 1280): Promise<string> {
+    try {
+      const resp = await this.ctx.http(url, { method: 'GET', responseType: 'arraybuffer', proxyAgent: '' })
+      let buffer = Buffer.from(resp.data)
+      let contentType = resp.headers.get('content-type')
+      if (resize) {
+        let img = sharp(buffer)
+        const metadata = await img.metadata()
+        const ratio = size / Math.max(metadata.width, metadata.height)
+        if(ratio < 1){
+          img = img.resize({
+            width: Math.floor(metadata.width * ratio),
+            height: Math.floor(metadata.height * ratio),
+          })
```

Comment:
Do not delegate other error to the network layer; they should be handled separately. You are trying to catch errors here that do not originate from the network request layer.